### DISCUSSION TITLE
Guava osv.dev CVE causing a "Action Required: Fix Renovate Configuration" issue

### DISCUSSION THREAD

**Discussion by ipletnjov-tw** [renovatebot/renovate#29826] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, version 37.415.0

### Please tell us more about your question or problem

Reproduction repository: https://github.com/ipletnjov-tw/renovate-guava-osv-config-issue

A Java (Maven) repository containing an outdated Guava version combined with `"osvVulnerabilityAlerts": true` and a packageRule with `"versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(-(?<compatibility>.*))?$"` for the Guava package will result in a "**Action Required: Fix Renovate Configuration**" issue being created by Renovate.

Example from the repro repo:
* https://github.com/ipletnjov-tw/renovate-guava-osv-config-issue/pull/1
(Part of the onboarding PR for some reason, instead of an issue)

The regex versioning string above was taken directly from Renovate docs on `regex` versioning: 
https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/regex/readme.md?plain=1#L21

This seems to be a regression caused by https://github.com/renovatebot/renovate/pull/29666, as running Renovate version `37.408.1` auto-closed the "Action Required: Fix Renovate Configuration" issue in our repos and our PRs were updated as usual.

### Logs (if relevant)

<details><summary>Logs</summary>

```
Let me know if the repro repo isn't enough and I'll try and get my hands on the debug logs.
```

</details>

**Comment by ipletnjov-tw** [comment]
May be of interest: flipping `osvVulnerabilityAlerts` to `false` (https://github.com/ipletnjov-tw/renovate-guava-osv-config-issue/commit/66663c06f802749996aad5fe7af5b9ff7f9362e4) auto-closed the "Action Required: Fix Renovate Configuration" PR.

**Comment by viceice** [comment]
use version compability instead of regex versioning 

https://docs.renovatebot.com/configuration-options/#versioncompatibility

**Reply by ipletnjov-tw** [reply]
Thanks @viceice. This caused Renovate to auto-close both of the PRs in the repro repo: https://github.com/ipletnjov-tw/renovate-guava-osv-config-issue/pulls?q=is%3Apr+is%3Aclosed.

Is there something wrong with the regex that I used for Guava? `^(?<version>[^-]+)(?<compatibility>-.*)?$`

I would have expected at least an upgrade to 23.0 with this config.

**Reply by viceice** [reply]
don't know, check the debug logs

**Reply by rarkins** [reply]
Non remediation PRs are always auto closed to favor the remediation PRs

The problem here is that Regex versioning doesn't support >= syntax so we should change our recommendation to use extractVersion

**Reply by ipletnjov-tw** [reply, marked as answer]
For posterity, the following configuration is where I ended up with this:
```json5
{
  "packageRules": [
    {
      "matchCategories": ["java"],
      "matchPackageNames": ["com.google.guava:guava"],
      "matchCurrentValue": "!/-(jre|android)$/",
      "allowedVersions": "/-jre$/",
      "groupName": "all dependencies",
      "groupSlug": "all",
      "separateMajorMinor": false,
      "dependencyDashboardApproval": false,
      "enabled": true
    },
    // Guava dependencies suffixed with -jre or -android will keep their current compatibility suffix.
    {
      "matchCategories": ["java"],
      "matchPackageNames": ["com.google.guava:guava"],
      "matchCurrentValue": "/-(jre|android)$/",
      "versionCompatibility": "^(?<version>[^-]+)(?<compatibility>-.*)?$",
      "extractVersion": "^(?<version>.*)(-(?<compatibility>.*))?$",
      "groupName": "all dependencies",
      "groupSlug": "all",
      "separateMajorMinor": false,
      "dependencyDashboardApproval": false,
      "enabled": true
    }
  ]
}
```

This is a workaround at best, but it'll ensure all Guava versions are upgraded and will keep their current compatibility suffix. Unfortunately Guava versions old enough to be unsuffixed will be upgraded to `-android` instead of `-jre`.

Example PR:
* https://github.com/ipletnjov-tw/renovate-guava-osv-config-issue/pull/6

**Reply by ipletnjov-tw** [reply]
The above config got me "close enough" to the answer that I'll close this discussion.

I will leave someone else take a stab at the linked issue https://github.com/renovatebot/renovate/issues/29932, as I don't think the above is simple/efficient enough for a spot in your public docs.
